pkgs/build-support/rust/hooks/cargo-setup-hook.sh

```diff
@@ -36,6 +36,8 @@ cargoSetupPostUnpackHook() {
     @cargoConfig@
 EOF
 
+    substituteInPlace .cargo/config.toml --subst-var-by BUILD_TOP "${NIX_BUILD_TOP:-/build}"
```

Comment:
Should we not combine this with the `substitute` above?

Reply:
It's not quite that simple. The new flag is going in the `target.${rustcTarget}.rustflags` setting, which is specified in the `cargoConfig` value. That's not part of the contents being substituted by the `substitute` above.

The way to do this would be to rewrite the previous substitution entirely so it looks something like

```bash
    config="$cargoDepsCopy/.cargo/config.toml"
    if [[ ! -e $config ]]; then
      config=@defaultConfig@
    fi

    cat $config - >> .cargo/config.toml <<'EOF'
    @cargoConfig@
EOF

    substituteInPlace .cargo/config.toml \
      --subst-var-by vendor "$cargoDepsCopy" \
      --subst-var-by BUILD_TOP "${NIX_BUILD_TOP:-/build}"
```

Reply:
Either way, I suppose combining these into one is a decent idea. I'll sleep on it.